modules/effects/src/effects_resolver.ts

```diff
@@ -24,14 +24,19 @@ export function mergeEffects(
           ? sourceInstance[propertyName]()
           : sourceInstance[propertyName];
 
+      const resubscribeInCaseOfError = (
+        observable$: Observable<any>
```

Comment:
both `any` should be replaced with `T` to make sure that both Observables are of the same kind.